diff -ur vixie.orig/cron.8 vixie/cron.8
--- vixie.orig/cron.8	2005-09-01 20:26:31.000000000 +0200
+++ vixie/cron.8	2005-09-02 09:45:56.000000000 +0200
@@ -36,7 +36,7 @@
 starting it out of init.
 .PP
 .I Cron
-searches /var/cron/tabs for crontab files which are named after accounts in
+searches /var/spool/cron for crontab files which are named after accounts in
 /etc/passwd; crontabs found are loaded into memory.
 .I Cron
 also searches for /etc/crontab which is in a different format (see
diff -ur vixie.orig/crontab.1 vixie/crontab.1
--- vixie.orig/crontab.1	2005-09-01 20:26:31.000000000 +0200
+++ vixie/crontab.1	2005-09-02 09:45:56.000000000 +0200
@@ -88,8 +88,8 @@
 crontab(5), cron(8)
 .SH FILES
 .nf
-/var/cron/cron.allow
-/var/cron/cron.deny
+/etc/cron/cron.allow
+/etc/cron/cron.deny
 .fi
 .SH STANDARDS
 The
diff -ur vixie.orig/globals.h vixie/globals.h
--- vixie.orig/globals.h	2005-09-01 20:26:31.000000000 +0200
+++ vixie/globals.h	2005-09-02 09:45:56.000000000 +0200
@@ -59,7 +59,7 @@
 #endif
 	;
 
-XTRN char	*ProgramName INIT("amnesia");
+XTRN char	*ProgramName INIT("crond");
 XTRN int	LineNumber INIT(0);
 XTRN time_t	StartTime INIT(0);
 XTRN int	NoFork INIT(0);
diff -ur vixie.orig/Makefile vixie/Makefile
--- vixie.orig/Makefile	2005-09-01 20:26:31.000000000 +0200
+++ vixie/Makefile	2005-09-02 09:46:13.000000000 +0200
@@ -54,15 +54,15 @@
 DESTROOT	=	$(DESTDIR)/usr
 DESTSBIN	=	$(DESTROOT)/sbin
 DESTBIN		=	$(DESTROOT)/bin
-DESTMAN		=	$(DESTROOT)/share/man
+DESTMAN		=	$(DESTROOT)/man
 #<<need bitstring.h>>
 INCLUDE		=	-I.
 #INCLUDE	=
 #<<need getopt()>>
 LIBS		=
 #<<optimize or debug?>>
-#CDEBUG		=	-O
-CDEBUG		=	-g
+CDEBUG		=	$(RPM_OPT_FLAGS)
+#CDEBUG		=	-g
 #<<lint flags of choice?>>
 LINTFLAGS	=	-hbxa $(INCLUDE) $(DEBUGGING)
 #<<want to use a nonstandard CC?>>
@@ -75,7 +75,7 @@
 #INSTALL = installbsd
 INSTALL = install
 #<<any special load flags>>
-LDFLAGS		=
+LDFLAGS		=	-s
 #################################### end configurable stuff
 
 SHELL		=	/bin/sh
@@ -110,23 +110,26 @@
 			$(CC) $(LDFLAGS) -o crontab $(CRONTAB_OBJ) $(LIBS)
 
 install		:	all
-			$(INSTALL) -c -m  111 -o root -s cron    $(DESTSBIN)/
-			$(INSTALL) -c -m 4111 -o root -s crontab $(DESTBIN)/
-#			$(INSTALL) -c -m  111 -o root -g crontab -s cron $(DESTSBIN)/
-#			$(INSTALL) -c -m 2111 -o root -g crontab -s crontab $(DESTBIN)/
+			$(INSTALL) -d $(DESTSBIN) $(DESTBIN) $(DESTMAN)/man1 $(DESTMAN)/man8 $(DESTMAN)/man5
+			$(INSTALL) -c -m 755 cron $(DESTSBIN)/crond
+			$(INSTALL) -c -m 755 -s crontab $(DESTBIN)/
 			sh putman.sh crontab.1 $(DESTMAN)
+			chmod 644 $(DESTMAN)/man1/crontab.1
 			sh putman.sh cron.8    $(DESTMAN)
+			chmod 644 $(DESTMAN)/man8/cron.8
+			echo ".so cron.8" > $(DESTMAN)/man8/crond.8
 			sh putman.sh crontab.5 $(DESTMAN)
+			chmod 644 $(DESTMAN)/man5/crontab.5
 
 distclean	:	clean
 			rm -f *.orig *.rej *.BAK *.CKP *~ #*
 			rm -f a.out core tags
 
 clean		:
-			rm -f *.o
+			rm -f *.o a.out core tags *~ #*
 			rm -f cron crontab
 
-tags		:;	ctags ${SOURCES}
+tags		:	ctags ${SOURCES}
 
 kit		:	$(SHAR_SOURCE)
 			shar $(SHAR_SOURCE) >kit
diff -ur vixie.orig/pathnames.h vixie/pathnames.h
--- vixie.orig/pathnames.h	2005-09-01 20:26:31.000000000 +0200
+++ vixie/pathnames.h	2005-09-02 09:45:56.000000000 +0200
@@ -35,7 +35,7 @@
 			 * to; SPOOL_DIR, CRON_ALLOW, CRON_DENY, and LOG_FILE
 			 * are all relative to this directory.
 			 */
-#define CRONDIR		"/var/cron"
+#define CRONDIR		"/var/spool"
 #endif
 
 			/* SPOOLDIR is where the crontabs live.
@@ -46,13 +46,13 @@
 			 * newer than they were last time around (or which
 			 * didn't exist last time around...)
 			 */
-#define SPOOL_DIR	"tabs"
+#define SPOOL_DIR	"cron"
 
 			/* cron allow/deny file.  At least cron.deny must
 			 * exist for ordinary users to run crontab.
 			 */
-#define	CRON_ALLOW	"cron.allow"
-#define	CRON_DENY	"cron.deny"
+#define	CRON_ALLOW	"/etc/cron/cron.allow"
+#define	CRON_DENY	"/etc/cron/cron.deny"
 
 			/* undefining this turns off logging to a file.  If
 			 * neither LOG_FILE or SYSLOG is defined, we don't log.
@@ -60,7 +60,7 @@
 			 * LOG_CRON is defined by <syslog.h>, LOG_FILE will not
 			 * be used.
 			 */
-#define LOG_FILE	"log"
+#define LOG_FILE	"/var/log/cron"
 
 			/* where should the daemon stick its PID?
 			 * PIDDIR must end in '/'.
@@ -70,7 +70,7 @@
 #else
 # define PIDDIR "/etc/"
 #endif
-#define PIDFILE		"cron.pid"
+#define PIDFILE		"crond.pid"
 #define _PATH_CRON_PID	PIDDIR PIDFILE
 
 			/* 4.3BSD-style crontab */
