diff -urN vixie-cron-3.0.1-orig/cron.c vixie-cron-3.0.1/cron.c
--- vixie-cron-3.0.1-orig/cron.c	1995-05-31 23:37:20.000000000 +0200
+++ vixie-cron-3.0.1/cron.c	2005-05-30 14:47:50.437022176 +0200
@@ -239,7 +239,8 @@
 	while (seconds_to_wait > 0) {
 		Debug(DSCH, ("[%d] sleeping for %d seconds\n",
 			getpid(), seconds_to_wait))
-		seconds_to_wait = (int) sleep((unsigned int) seconds_to_wait);
+                sleep((unsigned int) seconds_to_wait);
+                seconds_to_wait = (int) (TargetTime - time((time_t*) 0));
 	}
 }
 
