--- vixie-cron-3.0.1/cron.h.name	Fri Jan 19 11:35:25 2001
+++ vixie-cron-3.0.1/cron.h	Fri Jan 19 11:35:47 2001
@@ -253,6 +253,7 @@
 	};
 
 char	*ProgramName;
+char    *SyslogName;
 int	LineNumber;
 time_t	TargetTime;
 
@@ -267,7 +268,8 @@
 extern	char	*copyright[],
 		*MonthNames[],
 		*DowNames[],
-		*ProgramName;
+		*ProgramName,
+                *SyslogName;
 extern	int	LineNumber;
 extern	time_t	TargetTime;
 # if DEBUGGING
--- vixie-cron-3.0.1/do_command.c.name	Fri Jan 19 11:36:22 2001
+++ vixie-cron-3.0.1/do_command.c	Fri Jan 19 11:36:04 2001
@@ -86,6 +86,7 @@
 	/*local*/{
 		register char	*pch;
 
+		SyslogName = strdup(ProgramName);
 		for (pch = ProgramName;  *pch;  pch++)
 			*pch = MkUpper(*pch);
 	}
--- vixie-cron-3.0.1/misc.c.name	Fri Jan 19 11:36:28 2001
+++ vixie-cron-3.0.1/misc.c	Fri Jan 19 11:36:43 2001
@@ -522,9 +522,9 @@
 		 * print the pid ourselves.
 		 */
 # ifdef LOG_DAEMON
-		openlog(ProgramName, LOG_PID, LOG_CRON);
+		openlog(SyslogName, LOG_PID, LOG_CRON);
 # else
-		openlog(ProgramName, LOG_PID);
+		openlog(SyslogName, LOG_PID);
 # endif
 		syslog_open = TRUE;		/* assume openlog success */
 	}
