diff -uNr vixie-cron-4.1.p6/do_command.c vixie-cron-4.1.p7/do_command.c
--- vixie-cron-4.1.p6/do_command.c	2005-07-24 21:57:56.000000000 +0200
+++ vixie-cron-4.1.p7/do_command.c	2005-07-25 00:10:45.000000000 +0200
@@ -265,6 +265,14 @@
 				_exit(OK_EXIT);
 			}
 # endif /*DEBUGGING*/
+#ifdef USE_SIGCHLD
+			/* Our grandparent is watching for our parent's death by
+			 * catching SIGCHLD. Meanwhile, our parent will use wait
+			 * explicitly and so has disabled SIGCHLD. So now it's
+			 * time to reset SIGCHLD handling.
+			 */
+			(void) signal(SIGCHLD, SIG_DFL);
+#endif /*USE_SIGCHLD*/
 			execle(shell, shell, "-c", e->cmd, (char *)0, e->envp);
 			fprintf(stderr, "execl: couldn't exec `%s'\n", shell);
 			perror("execl");
