diff -uNr vixie-cron-4.1.p9/do_command.c vixie-cron-4.1/do_command.c
--- vixie-cron-4.1.p9/do_command.c	2005-07-24 22:23:10.000000000 +0200
+++ vixie-cron-4.1/do_command.c	2005-07-24 23:13:19.000000000 +0200
@@ -75,6 +75,7 @@
 	/*local*/{
 		char	*pch;
 
+		SyslogName = strdup(ProgramName);
 		for (pch = ProgramName;  *pch;  pch++)
 			*pch = MkUpper(*pch);
 	}
diff -uNr vixie-cron-4.1.p9/globals.h vixie-cron-4.1/globals.h
--- vixie-cron-4.1.p9/globals.h	2005-07-23 12:40:30.000000000 +0200
+++ vixie-cron-4.1/globals.h	2005-07-24 23:11:28.000000000 +0200
@@ -60,6 +60,7 @@
 	;
 
 XTRN char	*ProgramName INIT("crond");
+XTRN char	*SyslogName INIT("crond");
 XTRN int	LineNumber INIT(0);
 XTRN time_t	StartTime INIT(0);
 XTRN int	NoFork INIT(0);
diff -uNr vixie-cron-4.1.p9/misc.c vixie-cron-4.1/misc.c
--- vixie-cron-4.1.p9/misc.c	2005-07-24 22:15:24.000000000 +0200
+++ vixie-cron-4.1/misc.c	2005-07-24 23:14:19.000000000 +0200
@@ -521,9 +521,9 @@
 #if defined(SYSLOG)
 	if (!syslog_open) {
 # ifdef LOG_DAEMON
-		openlog(ProgramName, LOG_PID, FACILITY);
+		openlog(SyslogName, LOG_PID, FACILITY);
 # else
-		openlog(ProgramName, LOG_PID);
+		openlog(SyslogName, LOG_PID);
 # endif
 		syslog_open = TRUE;		/* assume openlog success */
 	}
