--- vixie-cron-3.0.1/compat.h.linux	Tue Oct  5 11:02:06 1999
+++ vixie-cron-3.0.1/compat.h	Tue Oct  5 11:02:39 1999
@@ -62,7 +62,7 @@
 #endif
 
 #ifndef POSIX
-# if (BSD >= 199103) || defined(__linux) || defined(ultrix) || defined(AIX) ||\
+# if (BSD >= 199103) || defined(__linux__) || defined(ultrix) || defined(AIX) ||\
 	defined(HPUX) || defined(CONVEX) || defined(IRIX)
 #  define POSIX
 # endif
@@ -76,16 +76,16 @@
 
 /*****************************************************************/
 
-#if !defined(BSD) && !defined(HPUX) && !defined(CONVEX) && !defined(__linux)
+#if !defined(BSD) && !defined(HPUX) && !defined(CONVEX) && !defined(__linux__)
 # define NEED_VFORK
 #endif
 
-#if (!defined(BSD) || (BSD < 198902)) && !defined(__linux) && \
+#if (!defined(BSD) || (BSD < 198902)) && !defined(__linux__) && \
 	!defined(IRIX) && !defined(NeXT) && !defined(HPUX)
 # define NEED_STRCASECMP
 #endif
 
-#if (!defined(BSD) || (BSD < 198911)) && !defined(__linux) &&\
+#if (!defined(BSD) || (BSD < 198911)) && !defined(__linux__) &&\
 	!defined(IRIX) && !defined(UNICOS) && !defined(HPUX)
 # define NEED_STRDUP
 #endif
@@ -102,7 +102,7 @@
 # define NEED_SETSID
 #endif
 
-#if (defined(POSIX) && !defined(BSD)) && !defined(__linux)
+#if (defined(POSIX) && !defined(BSD)) && !defined(__linux__)
 # define NEED_GETDTABLESIZE
 #endif
 
