--- vixie-cron-3.0.1/database.c.orig	2003-04-23 22:28:41.000000000 +0200
+++ vixie-cron-3.0.1/database.c	2003-04-23 22:33:41.000000000 +0200
@@ -113,7 +113,7 @@
 		 * to getpwnam() for . and .., and there shouldn't be 
 		 * hidden files in here anyway
 		 */
-		if (dp->d_name[0] == '.')
+		if (dp->d_name[0] == '.' && dp->d_name[strlen(dp->d_name)-1] != '~')
 			continue;
 		/* ignore files starting with # and ending with ~ */
 
