diff -ur vixie.orig/crontab.c vixie/crontab.c
--- vixie.orig/crontab.c	2005-09-01 20:26:31.000000000 +0200
+++ vixie/crontab.c	2005-09-01 22:34:16.000000000 +0200
@@ -189,7 +189,9 @@
 			Option = opt_replace;
 			if (strlen(argv[optind]) >= sizeof Filename)
 				usage("filename too long");
-			(void) strcpy (Filename, argv[optind]);
+			(void) strncpy (Filename, argv[optind],
+					sizeof(Filename) - 1);
+			Filename[sizeof(Filename) - 1] = '\0';
 		} else
 			usage("file name must be specified for replace");
 	}
diff -ur vixie.orig/env.c vixie/env.c
--- vixie.orig/env.c	2005-09-01 20:26:31.000000000 +0200
+++ vixie/env.c	2005-09-01 22:38:04.000000000 +0200
@@ -139,8 +139,9 @@
 	filepos = ftell(f);
 	fileline = LineNumber;
 	skip_comments(f);
-	if (EOF == get_string(envstr, MAX_ENVSTR, f, "\n"))
+	if (EOF == get_string(envstr, MAX_ENVSTR - 1, f, "\n"))
 		return (ERR);
+	envstr[MAX_ENVSTR - 1] = '\0';
 
 	Debug(DPARS, ("load_env, read <%s>\n", envstr))
 
