diff -Nru vixie-cron-3.0.1/Makefile vixie-cron-3.0.1.new/Makefile
--- vixie-cron-3.0.1/Makefile	Sun Oct 10 20:34:33 1999
+++ vixie-cron-3.0.1.new/Makefile	Sun Oct 10 20:38:27 1999
@@ -113,19 +113,16 @@
 			$(CC) $(LDFLAGS) -o crontab $(CRONTAB_OBJ) $(LIBS)
 
 install		:	all
-			$(INSTALL) -c -m 755 -o root -s cron    $(DESTSBIN)/crond
-			$(INSTALL) -c -m 4755 -o root -s crontab $(DESTBIN)/
+			$(INSTALL) -d $(DESTSBIN)/crond $(DESTBIN) $(DESTMAN)/man1 $(DESTMAN)/man8 $(DESTMAN)/man5
+			$(INSTALL) -c -m 755 $(DESTSBIN)/crond
+			$(INSTALL) -c -m 755 -s crontab $(DESTBIN)/
 			sh putman.sh crontab.1 $(DESTMAN)
 			chmod 644 $(DESTMAN)/man1/crontab.1
-			chown root.root $(DESTMAN)/man1/crontab.1
 			sh putman.sh cron.8    $(DESTMAN)
 			chmod 644 $(DESTMAN)/man8/cron.8
-			chown root.root $(DESTMAN)/man8/cron.8
-			rm -f $(DESTMAN)/man8/crond.8
-			ln -sf cron.8 $(DESTMAN)/man8/crond.8
+			echo ".so cron.8" > $(DESTMAN)/man8/crond.8
 			sh putman.sh crontab.5 $(DESTMAN)
 			chmod 644 $(DESTMAN)/man5/crontab.5
-			chown root.root $(DESTMAN)/man5/crontab.5
 
 clean		:
 			rm -f *.o cron crontab a.out core tags *~ #*
