diff -ur vixie.orig/config.h vixie/config.h
--- vixie.orig/config.h	2005-08-03 14:22:29.000000000 +0200
+++ vixie/config.h	2005-08-03 14:41:00.745117840 +0200
@@ -29,7 +29,7 @@
  */
 
 #ifndef DEBUGGING
-#define DEBUGGING 1	/* 1 or 0 -- do you want debugging code built in? */
+#define DEBUGGING 0	/* 1 or 0 -- do you want debugging code built in? */
 #endif
 
 			/*
diff -ur vixie.orig/cron.c vixie/cron.c
--- vixie.orig/cron.c	2005-08-03 14:22:29.000000000 +0200
+++ vixie/cron.c	2005-08-03 14:42:01.843829424 +0200
@@ -48,10 +48,13 @@
 usage(void) {
 	const char **dflags;
 
-	fprintf(stderr, "usage:  %s [-f] [-n] [-x [", ProgramName);
+	fprintf(stderr, "usage:  %s [-f] [-n]", ProgramName);
+#if DEBUGGING
+	fprintf(stderr, " [-x [");
 	for (dflags = DebugFlagNames; *dflags; dflags++)
-		fprintf(stderr, "%s%s", *dflags, dflags[1] ? "," : "]");
-	fprintf(stderr, "]\n");
+		fprintf(stderr, "%s%s", *dflags, dflags[1] ? "," : "]]");
+#endif
+	fprintf(stderr, "\n");
 	exit(ERROR_EXIT);
 }
 
