--- vixie-cron-3.0.1/config.h.notdot	Sun Jun 11 13:15:55 2000
+++ vixie-cron-3.0.1/config.h	Sun Jun 11 13:16:20 2000
@@ -42,8 +42,9 @@
 			 */
 
 #define MAILCMD _PATH_SENDMAIL					/*-*/
-#define MAILARGS "%s -FCronDaemon -odi -oem %s"			/*-*/
+#define MAILARGS "%s -FCronDaemon -i -odi -oem %s"			/*-*/
 			/* -Fx	 = set full-name of sender
+			 * -i	 = ignore single dot on line
 			 * -odi	 = Option Deliverymode Interactive
 			 * -oem	 = Option Errors Mailedtosender
 			 * -or0s = Option Readtimeout -- don't time out
